enginetest/join_op_tests.go

```diff
@@ -2275,6 +2275,29 @@ WHERE
 			},
 		},
 	},
+	{
+		// https://github.com/dolthub/dolt/issues/10233
+		name: "inner join with out of range key",
+		setup: [][]string{
+			{
+				"create table t0(c0 boolean, c1 int)",
+				"create index t0i0 on t0(c0, c1)",
+				"create table t1(c0 int)",
+				"insert into t0(c0, c1) values (-128, 1)",
+				"insert into t1(c0) values (1),(2)",
+			},
+		},
+		tests: []JoinOpTests{
+			{
+				Query:    "select * from t1 inner join t0 on t0.c0<=>(-87840)",
```

Comment:
Should probably have a version of this assertion with a join hint to force a lookup join if that's what you're targeting